lib/pure/asyncfile.nim

```diff
@@ -454,8 +454,15 @@ proc write*(f: AsyncFile, data: string): Future[void] =
 
     proc cb(fd: AsyncFD): bool =
       result = true
+
       let remainderSize = data.len-written
-      let res = write(fd.cint, addr copy[written], remainderSize.cint)
+
+      let res =
+        if data.len == 0:
+          0
+        else:
+          write(fd.cint, addr copy[written], remainderSize.cint)
```

Comment:
Please don't change this path. POSIX does permit actions to be taken by the operating system even when `0` is the length.

Reply:
I will chnage to:
```nim
if data.len == 0:
  write(fd.cint, copy.cstring, 0)
```